agent/consul/state/config_entry.go

```diff
@@ -92,10 +92,10 @@ func (s *Store) ConfigEntry(ws memdb.WatchSet, kind, name string) (uint64, struc
 	if err != nil {
 		return 0, nil, fmt.Errorf("failed config entry lookup: %s", err)
 	}
+	ws.Add(watchCh)
```

Comment:
Had to move this so blocking queries starter prior to the item existing would be woken up once one was added.